### PR TITLE
313 test

### PR DESCRIPTION
**Pull request by GregBestland**
Adding some very basic tests for 313

### CODE REVIEW

File: tests/integration/standard/test_custom_protocol_handler.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    import unittest2 as unittest
+except ImportError:
+    import unittest  # noqa
+
+from cassandra.protocol import ProtocolHandler, ResultMessage, UUIDType, read_int, EventMessage
+from cassandra.query import tuple_factory
+from cassandra.cluster import Cluster
+from tests.integration import use_singledc, PROTOCOL_VERSION, execute_until_pass
+from tests.integration.datatype_utils import update_datatypes, PRIMITIVE_DATATYPES, get_sample
+import time
```
Comment (aholmberg) [inline tests/integration/standard/test_custom_protocol_handler.py]: Unused

File: tests/integration/standard/test_custom_protocol_handler.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    @classmethod
+    def setUpClass(cls):
+        cls.cluster = Cluster(protocol_version=PROTOCOL_VERSION)
+        cls.session = cls.cluster.connect()
+        cls.session.execute("CREATE KEYSPACE custserdes WITH replication = { 'class' : 'SimpleStrategy', 'replication_factor': '1'}")
+        cls.session.set_keyspace("custserdes")
+
+    @classmethod
+    def tearDownClass(cls):
+        execute_until_pass(cls.session, "DROP KEYSPACE custserdes")
```
Comment (aholmberg) [inline tests/integration/standard/test_custom_protocol_handler.py]: Does this actually fail sometimes? :-S

File: tests/integration/standard/test_custom_protocol_handler.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        cls.cluster.shutdown()
+
+    def test_custom_raw_uuid_row_results(self):
+        """
+        Test to validate that custom protocol handlers work with raw row results
+
+        Connect and validate that the normal protocol handler is used.
+        Re-Connect and validate that the custom protocol handler is used.
```
Comment (aholmberg) [inline tests/integration/standard/test_custom_protocol_handler.py]: I think with the API changes we do not need to reconnect Clusters/Sessions. We can just replace the ``session.client_protocol_handler`` for each phase.

File: tests/integration/standard/test_custom_protocol_handler.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        result_set = session.execute("SELECT schema_version FROM system.local")
+        result = result_set.pop()
+        uuid_type = result[0]
+        self.assertEqual(type(uuid_type), uuid.UUID)
+        session.shutdown()
+
+        # Connect using our custom protocol handler, make sure we get raw data back
+        s = Cluster().connect()
+        s.client_protocol_handler = CustomTestRawRowType
+        session.client_protocol_handler = CustomTestRawRowType
```
Comment (aholmberg) [inline tests/integration/standard/test_custom_protocol_handler.py]: Not sure what we're doing with two sessions here. Maybe this gets cleaned up based on comment above.

File: tests/integration/standard/test_custom_protocol_handler.py (hunk 2 above)
Comment (GregBestland) [inline tests/integration/standard/test_custom_protocol_handler.py]: In this case likely not.

**Comment by GregBestland** [comment]
Changes made based on review everything should be addressed.

**Pull request merged**